stored_requests/backends/http_fetcher/fetcher.go

```diff
@@ -80,8 +81,48 @@ func (fetcher *HttpFetcher) FetchRequests(ctx context.Context, requestIDs []stri
 	return
 }
 
-func (fetcher *HttpFetcher) FetchCategories(primaryAdServer, publisherId, iabCategory string) (string, error) {
-	return "", nil
+func (fetcher *HttpFetcher) FetchCategories(ctx context.Context, primaryAdServer, publisherId, iabCategory string) (string, error) {
+	if fetcher.Categories == nil {
+		fetcher.Categories = make(map[string]map[string]stored_requests.Category)
+	}
+
+	//in NewFetcher function there is a code to add "?" at the end of url
+	//in case of categories we don't expect to have any parameters, that's why we need to remove "?"
+	url := fmt.Sprintf("%s/%s.json", strings.Replace(fetcher.Endpoint, "?", "", -1), primaryAdServer)
+
+	dataName := primaryAdServer
+
+	if publisherId != "" {
+		dataName = fmt.Sprintf("%s_%s", primaryAdServer, publisherId)
+		url = fmt.Sprintf("%s/%s/%s.json", strings.Replace(fetcher.Endpoint, "?", "", -1), primaryAdServer, publisherId)
+	}
+
+	if data, ok := fetcher.Categories[dataName]; ok {
+		return data[iabCategory].Id, nil
```

Comment:
Overall the PR LGTM. I just have one small question though: Are we sure that the `data` will always have a value for `iabCategory`? Should we rather just do a small check before we return? Both here and towards the end of the function.

Reply:
@mansinahar yes, it's good to have this check. Thank you!

Reply:
Thanks @Kalypsonika !